src/main/java/com/google/maps/model/PlacesSearchResult.java

```diff
@@ -67,7 +67,7 @@
   /**
    * types contains an array of feature types describing the given result.
    */
-  public String types[];
+  public AddressType types[];
```

Comment:
The magic responsible for this is https://github.com/googlemaps/google-maps-services-java/blob/master/src/main/java/com/google/maps/internal/SafeEnumAdapter.java#L64

However, I'm having to revert this change because types on place search results aren't AddressTypes. They just look like them sometimes.